src/collocation.jl

```diff
@@ -10,7 +10,16 @@ end
 function BVPSystem(fun, bc, p, x, y, order)
     T, U = eltype(x), eltype(y)
     M, N = size(y)
-    BVPSystem{T,U}(order, M, N, fun, bc, p, x, y, vector_alloc(T, M, N), vector_alloc(T, M, N), eltype(y)(M))
+    BVPSystem{T,U}(order, M, N, fun, bc, p, x, y, vector_alloc(U, M, N), vector_alloc(U, M, N), typeof(x)(M))
```

Comment:
This is not right. It should be `vector_alloc(T, M, N)`. `U` is `Vector{T}`.

Reply:
Oh now I see, this I messed up, I'll undo this.